virtwho/parser.py

```diff
@@ -9,116 +31,267 @@
+                             % log.DEFAULT_LOG_DIR)
+    parser.add_argument("-l", "--log-dir", action="store", dest="log_dir", default=log.DEFAULT_LOG_DIR,
+                        help="The absolute path of the directory to log to. (Default '%s')" % log.DEFAULT_LOG_DIR)
+    parser.add_argument("-f", "--log-file", action="store", dest="log_file", default=log.DEFAULT_LOG_FILE,
+                        help="The file name to write logs to. (Default '%s')" % log.DEFAULT_LOG_FILE)
+    parser.add_argument("-r", "--reporter-id", action="store", dest="reporter_id", default=NotSetSentinel(),
+                        help="Label host/guest associations obtained by this instance of virt-who with the provided id.")
+
+    virt_group = parser.add_argument_group(
+        title="Virtualization backend",
+        description="Choose virtualization backend that should be used to gather host/guest associations"
+    )
+    virt_group.add_argument("--libvirt", action=StoreVirtType, dest="virtType", const="libvirt",
+                            default=None, help="Use libvirt to list virtual guests")
+    virt_group.add_argument("--vdsm", action=StoreVirtType, dest="virtType", const="vdsm",
+                            help="Use vdsm to list virtual guests")
+    virt_group.add_argument("--esx", action=StoreVirtType, dest="virtType", const="esx",
+                            help="Register ESX machines using vCenter")
+    virt_group.add_argument("--xen", action=StoreVirtType, dest="virtType", const="xen",
+                            help="Register XEN machines using XenServer")
+    virt_group.add_argument("--rhevm", action=StoreVirtType, dest="virtType", const="rhevm",
+                            help="Register guests using RHEV-M")
+    virt_group.add_argument("--hyperv", action=StoreVirtType, dest="virtType", const="hyperv",
+                            help="Register guests using Hyper-V")
+
+    manager_group = parser.add_argument_group(
+        title="Subscription manager",
+        description="Choose where the host/guest associations should be reported"
+    )
+    manager_group.add_argument("--sam", action="store_const", dest="smType", const=SAT6, default=SAT6,
+                               help="Report host/guest associations to the Subscription Asset Manager [default]")
+    manager_group.add_argument("--satellite6", action="store_const", dest="smType", const=SAT6,
+                               help="Report host/guest associations to the Satellite 6 server")
+    manager_group.add_argument("--satellite5", action="store_const", dest="smType", const=SAT5,
+                               help="Report host/guest associations to the Satellite 5 server")
+    manager_group.add_argument("--satellite", action="store_const", dest="smType", const=SAT5)
+
+    # FIXME: Remove all options of virtualization backend. Adding this wasn't happy design decision.
```

Comment:
Agreed, I think we should do this sooner rather than later. (Non-nacking comment)